lib/instrumentation/undici.js

```diff
@@ -116,22 +116,33 @@ function addDTHeaders({ transaction, config, request }) {
  */
 function createExternalSegment({ shim, request, parentSegment }) {
   const url = new URL(request.origin + request.path)
-  const name = NAMES.EXTERNAL.PREFIX + url.host + url.pathname
+  const obfuscatedPath = urltils.obfuscatePath(shim.agent.config, url.pathname)
+  const name = NAMES.EXTERNAL.PREFIX + url.host + obfuscatedPath
   // Metrics for `External/<host>` will have a suffix of undici
   // We will have to see if this matters for people only using fetch
   // It's undici under the hood so ¯\_(ツ)_/¯
   const segment = shim.createSegment(name, recordExternal(url.host, 'undici'), parentSegment)
+
+  // brown-fields alert!!
+  // instrumentation expects query params to be an object. url.searchParams is an Iterable
+  // convert to object
+  const queryParams = {}
+  url.searchParams.forEach((value, key) => {
+    queryParams[key] = value
+  })
```

Comment:
😕

Then whatever we do here is still going to be some looping in a potential hot path. But we can push the looping into the optimized JavaScript methods:

```suggestion
  const queryParams = Object.fromEntries(url.searchParams.entries())
```

Reply:
I can change but we're talking about query params, which is a very limited amount.  this previous code did the same thing with iterating via forEach